File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        b=set()
        left=0
        maxx=0
        for i in range(len(s)):
            while s[i] in b:
                b.remove(s[left])
                left+=1
            b.add(s[i])
            maxx=max(maxx,i-left+1)
        return maxx    

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max=0;
        ArrayList<Character> arr=new ArrayList<>();
        for(char i:s.toCharArray()){
            while(arr.contains(i)){
                arr.remove(0);
            }
            arr.add(i);
            max=Math.max(max,arr.size());
        }
        return max;
    }
}